src/spatialdata/_core/spatialdata.py

```diff
@@ -434,21 +637,27 @@ def filter_by_coordinate_system(self, coordinate_system: str | list[str], filter
         if isinstance(coordinate_system, str):
             coordinate_system = [coordinate_system]
         for element_type, element_name, element in self._gen_elements():
-            transformations = get_transformation(element, get_all=True)
-            assert isinstance(transformations, dict)
-            for cs in coordinate_system:
-                if cs in transformations:
-                    if element_type not in elements:
-                        elements[element_type] = {}
-                    elements[element_type][element_name] = element
-                    element_paths_in_coordinate_system.append(element_name)
-
-        if filter_table:
-            table = _filter_table_by_coordinate_system(self.table, element_paths_in_coordinate_system)
+            if element_type != "tables":
+                transformations = get_transformation(element, get_all=True)
+                assert isinstance(transformations, dict)
+                for cs in coordinate_system:
+                    if cs in transformations:
+                        if element_type not in elements:
+                            elements[element_type] = {}
+                        elements[element_type][element_name] = element
+                        element_paths_in_coordinate_system.append(element_name)
+
+        # TODO: check whether full table dict should be returned or only those which annotate elements. Also check
+        # filtering with tables having potentially different keys.
```

Comment:
> Also check filtering with tables having potentially different keys.

I think this is fine, the filtering should be done independently of the table keys.